src/main/java/org/apache/sysml/hops/Hop.java

```diff
@@ -365,7 +365,7 @@ private void constructAndSetCheckpointLopIfRequired()
 			}
 			else //default case
 			{
-				et = ExecType.SPARK;
+				et = OptimizerUtils.getRemoteExecType();
```

Comment:
For Flink we should not construct Checkpoint lops while Flink does not support caching - see Matthias' remark on the PR.